### PR TITLE
add nei page and code rewrite for radhatch

### PR DESCRIPTION
**Pull request by GlodBlock**
add the nei page for radhatch material:
image
display current mass and sievert in hatch:
image

have sound effects when running.
rewrite the container and check logic.
the gtpp compact is removed now, it will be put in gtpp itself.

### CODE REVIEW

File: src/main/java/com/github/bartimaeusnek/bartworks/neiHandler/IMCForNEI.java [hunk 1]
```diff
@@ -10,6 +10,19 @@ public static void IMCSender() {
         sendCatalyst("gt.recipe.craker", "gregtech:gt.blockmachines:13367", -10);
         sendHandler("bw.recipe.htgr", "gregtech:gt.blockmachines:12791");
         sendCatalyst("bw.recipe.htgr", "gregtech:gt.blockmachines:12791");
+        sendCatalyst("bw.recipe.radhatch", "gregtech:gt.blockmachines:12713");
```
Comment (miozune) [inline src/main/java/com/github/bartimaeusnek/bartworks/neiHandler/IMCForNEI.java:14]: nit: in the order of handler -> catalyst

File: src/main/java/com/github/bartimaeusnek/bartworks/server/container/GT_Container_RadioHatch.java [hunk 2]
```diff
@@ -64,88 +70,60 @@ public void detectAndSendChanges() {
...
+        this.buffer.put(0, mass);
+        this.buffer.putShort(Byte.BYTES, sv);
+        this.buffer.putShort(Byte.BYTES + Short.BYTES, sievert);
+        this.buffer.putShort(Byte.BYTES + Short.BYTES * 2, r);
+        this.buffer.putShort(Byte.BYTES + Short.BYTES * 3, g);
+        this.buffer.putShort(Byte.BYTES + Short.BYTES * 4, b);
+        this.buffer.putLong(Byte.BYTES + Short.BYTES * 5, teTimer);
+        this.buffer.putLong(Byte.BYTES + Short.BYTES * 5 + Long.BYTES, decayTime);
+        sendStateUpdate(clientHandle);
+        super.addCraftingToCrafters(clientHandle);
```
Comment (miozune) [inline src/main/java/com/github/bartimaeusnek/bartworks/server/container/GT_Container_RadioHatch.java]: probably call `super` first

File: src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/tiered/GT_MetaTileEntity_RadioHatch.java [hunk 3]
```diff
@@ -178,122 +162,56 @@ public void onPostTick(IGregTechTileEntity aBaseMetaTileEntity, long aTimer) {
...
-                            }
-                        }
-                        return false;
+                if (!this.lastFail && this.lastUsedItem != null && this.lastRecipe != null) {
+                    if (GT_Utility.areStacksEqual(this.lastUsedItem, lStack, true)) {
+                        this.mass = (byte) this.lastRecipe.mDuration;
+                        this.decayTime = this.lastRecipe.mSpecialValue;
+                        this.sievert = this.lastRecipe.mEUt;
+                        lStack.stackSize--;
+                        updateSlots();
```
Comment (miozune) [inline src/main/java/com/github/bartimaeusnek/bartworks/common/tileentities/tiered/GT_MetaTileEntity_RadioHatch.java:195]: need to update `material` as well here

**Review by miozune — COMMENTED** [review]
Adding helper method for `BWRecipes.instance.getMappingsFor(BWRecipes.RADHATCH).containsInput` would be good

**Comment by GlodBlock** [comment]
i got stuck with some irl issues now, so i will solve these requests days later

File: src/main/java/com/github/bartimaeusnek/bartworks/neiHandler/IMCForNEI.java (hunk 1 above)
Comment (GlodBlock) [reply src/main/java/com/github/bartimaeusnek/bartworks/neiHandler/IMCForNEI.java:14]: @miozune done now

**Review by miozune — COMMENTED** [review]
Code is ok, but hatch sound feels like noisy cicada

**Comment by GlodBlock** [comment]
do little troll

**Review by miozune — APPROVED** [review]
Well people will go muffler upgrade anyway

**Pull request merged**